### PR TITLE
[C#] remove enum from config and from the switch exercise

### LINKED ISSUE #2322: [C#] Improve Concept Exercise: enums. Cyclic Dependency Bug
This issue describes how to improve the `enums` concept exercise for the C# track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before creating the issue, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

Remove Cyclic Dependency Bug from C# concept exercises. 
The slug enums depends on switch-statements and the opposite too.

## Help

We could simplify the enums exercise to avoid using a switch statement, because the track has a dedicated switch-statements exercise.
Option 1: Keep current exercise but solve it with if/else
Option 2: Write a simpler enum for example converting ranges of integer to Grades: A/B/C; Priorities: Low, Medium, High; 

## Bug

The dashboard is broken when clicking on the tree view:
https://v3.exercism.io/csharp/maintaining/list
image

### PR DESCRIPTION
**Pull request by valentin-p**
Closes #2322

Let me know if anything need to change

### CODE REVIEW

**Comment by valentin-p** [comment]
- Removing the cyclic dependency
- Changing the enum to subclasses
- Changing the docs
- Changing the tests
- Adding my name in constributor

File: languages/csharp/exercises/concept/switch-statements/.docs/instructions.md [hunk 1]
```diff
@@ -26,31 +26,31 @@ PlayAnalyzer.AnalyzeOnField(10);
...
+Implement the `PlayAnalyzer.AnalyzeOffField()` method to output description of activities and characters around the field of play.
 
 You receive a stream of data that has been cleaned. Your task is to analyse it and output appropriate text to help the journalists.
 
 The data comprises:
 
 - shirt numbers (any `int`) -> text as per on field analysis
 - free form text (any `string`) -> the text unchanged
-- incidents in play (`Incident` enum) -> "RedCard", "Foul" etc.
+- incidents in play (any subclass of the type `Incident`) -> "Injury", "Foul" etc.
```
Comment (ErikSchierboom) [inline languages/csharp/exercises/concept/switch-statements/.docs/instructions.md:41]: As far as I can tell, we're not currently using the sub-classes anywhere in the example implementation. Maybe we should have separate cases for maybe one of the sub-classes and one case for the general base class?

**Review by ErikSchierboom — COMMENTED** [review]
Great work! I've posted some comments.

**Review by ErikSchierboom — APPROVED** [review]
LGTM! Let's see what @mikedamay thinks before merging.

**Review by mikedamay — APPROVED** [review]
It looks very good to me.

**Comment by ErikSchierboom** [comment]
Thanks @valentin-p! 🎉

**Pull request merged**